external/fv3fit/fv3fit/emulation/transforms/factories.py

```diff
@@ -148,3 +149,23 @@ def build(self, sample: TensorDict) -> ComposedTransform:
             sample.update(transform.forward(sample))
             transforms.append(transform)
         return ComposedTransform(transforms)
+
+
+@dataclasses.dataclass
+class EnforcePositiveVariables(ComposedTransformFactory):
+    """
+    A convenience factory to apply PositiveTransform to
+    multiple fields with a single configuration entrypoint.
+
+    Attributes:
+        enforce_positive_on: List of variable names to apply limiter to
+    """
+
+    enforce_positive_on: Sequence[str]
+
+    def __init__(self, enforce_positive_on: Sequence[str]):
```

Comment:
I think the inheritance makes this code a bit magical, and subject to breaking if the `.factories` attribute changes for some reason. It's also odd/not correct for a `dataclass` to have a __init__, which is why I suspect you needed to manually patch in the redundant `enforce_positive_on` attribute below. This really should just be a function returning an `TransformedVariableConfig`, but unfortunately that probably won't work with dacite.

Can you delete the class? The convenience it provides seems small relative to the maintenance burden, and we can already configure the positive enforcement explicitly using the `UnivariateTransform`.

Reply:
> It's also odd/not correct for a dataclass to have a init, which is why I suspect you needed to manually patch in the redundant enforce_positive_on attribute below.

Yeh, 😅, I after starting to write the init, I changed it to be just a class, but then I didn't think I could still use `dacite` to initialize it.   My main goal was to have something easy to configure here, but yeah if it's confusing/hard to maintain I can see why other suggestions would be preferable.